### PR TITLE
Make hostname renaming a smoother experience

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Fix the `spacewalk-hostname-rename` script to:
 * Avoid generating a new SSL CA if not needed
 * Allow using third-party SSL certificate
 * Refresh the minion pillar data since they also contain the server hostname in the repositories URLs

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- PR submitted: https://github.com/uyuni-project/uyuni-docs/pull/1158

- [X] **DONE**

## Test coverage
- Unit tests were added
- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/15801

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: utils/spacewalk-hostname-rename [hunk 1]
```diff
@@ -390,6 +426,25 @@ function re-generate_server_ssl_certificate {
...
+    for ID in `spacecmd -q api system.listSystemsWithEntitlement -- -A "salt_entitled" -F "%(id)s"`
+    do
+        if [ -n "$SIDS" ]
+        then
+            IDS="$SIDS,$ID"
+        else
+            IDS="$SID"
+        fi
+    done
+    SKIPPED=`spacecmd -q api system.refreshPillar -- -A "[\\\\\"general\\\\\",[$SIDS]]"`
```
Comment (mcalmer) [inline utils/spacewalk-hostname-rename:440]: Where does `$SIDS` come from?
I see only `ID` and `IDS` variable been set.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 2]
```diff
@@ -7771,6 +7794,69 @@ public int updatePackageState(User loggedInUser, Integer sid, String packageName
...
+     *
+     * @param loggedInUser The current user
+     * @param systemIds A list of systems ids to delete
+     * @return Returns the number of systems refreshed if successful
+     *
+     * @xmlrpc.doc refresh the pillar data of a list of systems. The subset value represents the pillar to be refreshed
+     * and can be one of 'general', 'group_membership', 'virtualization' or 'custom_info'.
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #array_single("int", "serverIds")
+     * @xmlrpc.returntype #array_single("int", "skippedIds", "System IDs which couldn't be refreshed")
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:7808]: nice catch! it returns skipped ones

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 3]
```diff
@@ -7771,6 +7794,69 @@ public int updatePackageState(User loggedInUser, Integer sid, String packageName
         return this.systemManager.retrieveSystemGroupsForSystemsWithEntitlementAndUser(loggedInUser, entitlement);
     }
 
+    /**
+     * Refresh all the pillar data of a list of systems.
+     *
+     * @param loggedInUser The current user
+     * @param systemIds A list of systems ids to delete
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:7801]: copy / paste issue, will fix. Thanks

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java [hunk 4]
```diff
@@ -7771,6 +7794,69 @@ public int updatePackageState(User loggedInUser, Integer sid, String packageName
...
+     *
+     * @param loggedInUser The current user
+     * @param subset the string representation of the pillar subset
+     * @param systemIds A list of systems ids to delete
+     * @return Returns the number of systems refreshed if successful
+     *
+     * @xmlrpc.doc refresh the pillar data of a list of systems. The subset value represents the pillar to be refreshed
+     * and can be one of 'general', 'group_membership', 'virtualization' or 'custom_info'.
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "subset", "subset of the pillar to refresh.")
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java:7825]: I documented them in the `@xmlrpc.doc` part

File: utils/spacewalk-hostname-rename (hunk 1 above)
Comment (cbosdo) [reply utils/spacewalk-hostname-rename:440]: `$IDS` is empty (undefined to be more precise) at the beginning of the loop. This helps handling properly the comma separation between values

Comment (cbosdo) [reply utils/spacewalk-hostname-rename:440]: oops... looks like an incomplete rename to make `IDS` more understandable

**Review by mcalmer — APPROVED** [review]
LGTM

**Pull request merged**